src/range_proof.rs

```diff
@@ -751,54 +750,26 @@ where
 
     // Helper function to decompress Li
     fn li_decompressed(&self) -> Result<Vec<P>, ProofError> {
-        if self.li.is_empty() {
-            Err(ProofError::InvalidArgument("Vector 'L' not assigned yet".to_string()))
-        } else {
-            let mut li = Vec::with_capacity(self.li.len());
-            for item in self.li.clone() {
-                li.push(item.decompress().ok_or_else(|| {
-                    ProofError::InvalidArgument(
-                        "An item in member 'L' was not the canonical encoding of a point".to_string(),
-                    )
-                })?)
-            }
-            Ok(li)
-        }
-    }
-
-    // Helper function to return compressed Li
-    fn li(&self) -> Result<Vec<P::Compressed>, ProofError> {
-        if self.li.is_empty() {
-            Err(ProofError::InvalidArgument("Vector 'L' not assigned yet".to_string()))
-        } else {
-            Ok(self.li.clone())
-        }
+        self.li
+            .iter()
+            .map(|p| {
+                p.decompress().ok_or(ProofError::InvalidArgument(
+                    "An item in member 'L' was not the canonical encoding of a point".to_string(),
+                ))
+            })
+            .collect::<Result<Vec<_>, ProofError>>()
     }
 
     // Helper function to decompress Ri
     fn ri_decompressed(&self) -> Result<Vec<P>, ProofError> {
-        if self.ri.is_empty() {
-            Err(ProofError::InvalidArgument("Vector 'R' not assigned yet".to_string()))
```

Comment:
ditto: I would not remove this safety check

Reply:
See above.